catalog/core/catalog-core-versioning/versioning-common/src/main/java/ddf/catalog/core/versioning/MetacardVersion.java

```diff
@@ -196,6 +236,20 @@ public MetacardVersion(Metacard sourceMetacard, Action action, Subject subject)
         this.setVersionOfId(sourceMetacard.getId());
         this.setVersionTags(sourceMetacard.getTags());
 
+        Optional<MetacardType> type = types.stream()
+                .filter(mt -> sourceMetacard.getMetacardType()
+                        .getName()
+                        .equals(mt.getName()))
+                .findFirst();
+
+        if (type.isPresent()) {
+            this.setVersionType(type.get()
```

Comment:
The result of setting the version type in the if & the else are the same- you could move it out for simplicity.